odl/tomo/geometry/parallel.py

```diff
@@ -343,76 +596,225 @@ class Parallel3dAxisGeometry(ParallelGeometry, AxisOrientedGeometry):
+        >>> np.allclose(geom.det_axes_init, (e_y, e_z))
+        True
+
+        Specifying an axis by default rotates the standard configuration
+        to this position:
+
+        >>> geom = Parallel3dAxisGeometry(apart, dpart, axis=(0, 1, 0))
+        >>> np.allclose(geom.axis, e_y)
+        True
+        >>> np.allclose(geom.det_pos_init, -e_x)
+        True
+        >>> np.allclose(geom.det_axes_init, (-e_z, e_y))
+        True
+        >>> geom = Parallel3dAxisGeometry(apart, dpart, axis=(1, 0, 0))
+        >>> np.allclose(geom.axis, e_x)
+        True
+        >>> np.allclose(geom.det_pos_init, e_z)
+        True
+        >>> np.allclose(geom.det_axes_init, (e_y, e_x))
+        True
+
+        The initial detector position and axes can also be set explicitly:
+
+        >>> geom = Parallel3dAxisGeometry(apart, dpart, axis=(0, 1, 0),
+        ...                               det_pos_init=(1, 0, 0))
+        >>> np.allclose(geom.axis, e_y)
+        True
+        >>> np.allclose(geom.det_pos_init, e_x)
+        True
+        >>> np.allclose(geom.det_axes_init, (-e_z, e_y))  # as above
+        True
+        >>> geom = Parallel3dAxisGeometry(
+        ...     apart, dpart, axis=(0, 1, 0),
+        ...     det_axes_init=((0, 1, 0), (0, 0, 1)))
+        >>> np.allclose(geom.axis, e_y)
+        True
+        >>> np.allclose(geom.det_pos_init, -e_x)
+        True
+        >>> np.allclose(geom.det_axes_init, (e_y, e_z))
+        True
+
+        A matrix can be given to perform a final rotation. This is most
+        useful to rotate non-standard ``det_axes_init``, or if full
+        control over the rotation is desired:
+
+        >>> rot_matrix = np.array([[1, 0, 0],
+        ...                        [0, 0, -1],
+        ...                        [0, 1, 0]])
+        >>> geom = Parallel3dAxisGeometry(apart, dpart, extra_rot=rot_matrix)
+        >>> np.allclose(geom.axis, -e_y)
+        True
+        >>> np.allclose(geom.det_pos_init, -e_x)  # default
+        True
+        >>> np.allclose(geom.det_axes_init, (e_z, -e_y))
+        True
         """
+        def_axis = self._default_config['axis']
+        def_det_pos = self._default_config['det_pos_init']
+        def_det_axes = self._default_config['det_axes_init']
+
+        det_pos_init = kwargs.pop('det_pos_init', None)
+        det_axes_init = kwargs.pop('det_axes_init', None)
+        extra_rot = np.asarray(kwargs.pop('extra_rot', np.eye(3)))
+
+        if extra_rot.shape != (3, 3):
+            raise ValueError('`extra_rot` must have shape (3, 3), got {}'
+                             ''.format(extra_rot.shape))
+        if abs(np.linalg.det(extra_rot)) < 1e-4:
+            raise ValueError('`extra_rot` is almost singular')
+        self.__extra_rotation = extra_rot
+
+        if kwargs:
+            raise TypeError('got an unexpected keyword argument {!r}'
+                            ''.format(kwargs.popitem()[0]))
+
+        if np.linalg.norm(np.asarray(axis) - def_axis) < 1e-4:
```

Comment:
How about actually checking the cosine distance here, that would be scale invariant?